### ISSUE TITLE
Support for optional properties and parameters

### ISSUE THREAD

**Issue by cayuu** [gajus/eslint-plugin-flowtype#27]
I probably missed a setting somewhere, but do optional props/params fly with arrow funcs?

```js
(optionalProp ?: string) => { (optionalProp : string|void) }
//             ^ Unexpected token (null)
```

Writing the above as a normal ES5 function validates fine. The pure "maybe" form seems to validate fine as an ES6 arrow func: 

```js
(maybe : ?string) => { (maybe : string|void|null) }
```

https://flowtype.org/docs/builtins.html#null-and-void

**Comment by emirotin** [comment]
Additionally: the `space-before-type-colon` rule doesn't seem to work with optional params.

**Comment by danharper** [comment]
Optional args are now supported as of #46

**Comment by gajus** [comment]
v2.4.2 has been released. Thanks to @danharper

**Comment by emirotin** [comment]
I'm using v2.11.0 and I've started getting the errors again, like:

> 5:35  warning  There must be a space after "i" parameter type annotation colon  flowtype/space-after-type-colon

referring to this line:

> type TArrayPredicate<T> = (el: T, i?: number) => boolean

**Comment by danharper** [comment]
Huh, thought I'd covered that case. There's a PR open for it now :)

**Comment by gajus** [comment]
v2.11.1 has been released. Thank you.

**Comment by emirotin** [comment]
Yeah, looks fine, thank you.
